dlp/api/Snippets/InspectStorageWithSampling.cs

```diff
@@ -0,0 +1,144 @@
+        // Construct storage config.
+        var storageConfig = new StorageConfig
+        {
+            CloudStorageOptions = new CloudStorageOptions
+            {
+                FileSet = new CloudStorageOptions.Types.FileSet
+                {
+                    Url = GcsFileUri
+                },
+                BytesLimitPerFile = 200,
+                FileTypes = { new FileType[] { FileType.Csv } },
+                FilesLimitPercent = 90,
+                SampleMethod = CloudStorageOptions.Types.SampleMethod.RandomStart
+            }
+        };
+
+        // Construct the inspect config.
+        var inspectConfig = new InspectConfig
+        {
+            InfoTypes =
+            {
+                infoTypes ?? new InfoType[] { new InfoType { Name = "PERSON_NAME" } }
+            },
+            IncludeQuote = true,
+            MinLikelihood = minLikelihood
+        };
+
+        // Construct the pubsub action.
+        var actions = new Google.Cloud.Dlp.V2.Action[]
+        {
+            new Google.Cloud.Dlp.V2.Action
+            {
+                PubSub = new Google.Cloud.Dlp.V2.Action.Types.PublishToPubSub
+                {
+                    Topic = $"projects/{projectId}/topics/{topicId}"
+                }
+            }
+        };
+
+        // Construct the inspect job config using actions.
+        var inspectJob = new InspectJobConfig
+        {
+            StorageConfig = storageConfig,
+            InspectConfig = inspectConfig,
+            Actions = { actions }
+        };
+
+        // Issue Create Dlp Job Request
+        var client = DlpServiceClient.Create();
+        var request = new CreateDlpJobRequest
+        {
+            InspectJob = inspectJob,
+            ParentAsLocationName = new LocationName(projectId, "global"),
+        };
+
+        // We need created job name
+        var dlpJob = dlp.CreateDlpJob(request);
+        var jobName = dlpJob.Name;
+
+        // Listen to pub/sub for the job
+        var subscriptionName = new SubscriptionName(projectId, subId);
+        var subscriber = SubscriberClient.CreateAsync(
+            subscriptionName).Result;
+
+        // SimpleSubscriber runs your message handle function on multiple threads to maximize throughput.
+        var done = new ManualResetEventSlim(false);
+        subscriber.StartAsync((PubsubMessage message, CancellationToken cancel) =>
+        {
+            if (message.Attributes["DlpJobName"] == jobName)
+            {
+                // Wait for DLP API results to become consistent
+                Thread.Sleep(500);
+                done.Set();
+                return Task.FromResult(SubscriberClient.Reply.Ack);
+            }
+            else
+            {
+                return Task.FromResult(SubscriberClient.Reply.Nack);
+            }
+        });
+
+        // 3 minute timeout; may not work for large jobs
+        done.Wait(TimeSpan.FromMinutes(3));
+        subscriber.StopAsync(CancellationToken.None).Wait();
```

Comment:
Please do not use Wait or Result as that blocks the thread and can lead to dealocks. This method should be async and tasks awaited.